js/exportsms.js

```diff
@@ -98,6 +99,94 @@ var exportsms = function() {
     return data;
   }
 
+  /**
+   * Get passage data out of passage elements and return as an object with just the info we need.
+   *
+   * @param passage
+   *   Passage data as an HTML tw-passagedata element
+   * @return Object
+   */
+  function _compilePassage(passage) {
+    var data = {};
+    var attrs = passage.attributes;
+
+    data.optinpath = attrs.getNamedItem('optinpath') ? attrs.getNamedItem('optinpath').value : 0;
+    data.name = attrs.getNamedItem('name') ? attrs.getNamedItem('name').value : '';
+    data.text = passage.innerText.trim();
+
+    return data;
+  }
+
+  /**
+   * Build story config out of an array of passage data.
+   *
+   * @param passages
+   *   Array of passages in the story
+   * @return Story object
+   */
+  function _buildStory(passages) {
+    var story,
+        storyPassage,
+        passage,
+        optinpath,
+        key,
+        links,
+        answers,
+        next,
+        error,
+        i,
+        j,
+        k;
+
+    story = {};
+    for (i = 0; i < passages.length; i++) {
+      passage = passages[i];
+      if (typeof story[passage.optinpath.toString()] === 'undefined') {
+        storyPassage = {};
+        storyPassage.name = passage.name;
+        storyPassage.choices = [];
+
+        // Find links in the text
+        links = passage.text.match(/\[\[.*?\]\]/g);
+
+        for (j = 0; links != null && j < links.length; j++) {
+          // Assumes link format is [[display text|link|valid answers]]
+          key = links[j].replace(/\[\[(.+)\|(.+)\|(.+)\]\]/g, '$2');
+          answers = links[j].replace(/\[\[(.+)\|(.+)\|(.+)\]\]/g, '$3');
+          next = 0;
+
+          // Build the choices array based on the links found
+          storyPassage.choices[j] = {};
+          storyPassage.choices[j].key = key;
+          storyPassage.choices[j].valid_answers = [answers];
+
+          // To get the `next` optinpath, search through all passages to find the
+          // passage.name that matches the key for this choice.
```

Comment:
ahh! Nice. This means that the built-in link generator of twinejs will be enough to visually validate that `next` values actually point to the right things.